spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala

```diff
@@ -1354,9 +1354,14 @@ object CometSparkSessionExtensions extends Logging {
 
   /** Calculates required memory overhead in MB per executor process for Comet. */
   def getCometMemoryOverheadInMiB(sparkConf: SparkConf): Long = {
-    // `spark.executor.memory` default value is 1g
-    val executorMemoryMiB = ConfigHelpers
-      .byteFromString(sparkConf.get("spark.executor.memory", "1024MB"), ByteUnit.MiB)
+    val executorMemoryMiB = if (cometUnifiedMemoryManagerEnabled(sparkConf)) {
```

Comment:
hmm it is a bit weird to put off heap memory amount to `executorMemoryMiB`

Reply:
> hmm it is a bit weird to put off heap memory amount to `executorMemoryMiB`

This is just a base value used to calculate `CometMemoryOverhead`, when we enable unifled memory manager we should calculate it from off-heap memory.

Reply:
I noticed this behavior from your previous PR, after this PR we don't need to configure manual `COMET_MEMORY_OVERHEAD` when using unified memory manager.

https://github.com/apache/datafusion-comet/pull/1369/files#diff-c104036e91cd748933259320a8d86535e5f6fccae24875daf0f18e6c5ca0ab64L96-L97

Reply:
I guess this could just be a naming issue. What about renaming `executorMemoryMiB` to `baseMemoryMiB ` as well as updating the description of `COMET_MEMORY_OVERHEAD_FACTOR` that currently says
`Fraction of executor memory`

Reply:
I am still confused here. For the unified memory manager, do we still need to multiply off-heap memory by a factor to calculate comet overhead memory? Or should we use executor off-heap memory as comet overhead memory?

Reply:
I guess the intention is

Off heap == unified memory
```
 - off-heap memory x factor
```

on heap == DataFusion memory pool
```
 - executor memory x factor
```
?